graalwasm/graalwasm-micronaut-excelice/src/main/java/com/example/Controller.java

```diff
@@ -0,0 +1,104 @@
+package com.example;
+
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.Part;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.multipart.CompletedFileUpload;
+import jakarta.inject.Inject;
+import io.micronaut.views.View;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+@io.micronaut.http.annotation.Controller
+public class Controller {
+
+    @Inject
```

Comment:
Please use [constructor injection](https://guides.micronaut.io/latest/micronaut-dependency-injection-types-maven-java.html#constructor-injection) here. It's not good to mix different injection kinds. Also, constructor injection allows you to mark the fields as final. Please do so.